From 03638140c122a946acafa46054a81315fa7709ce Mon Sep 17 00:00:00 2001
From: "wenxin.leong" <wenxin.leong@infineon.com>
Date: Sun, 7 Feb 2021 21:57:15 -0800
Subject: [PATCH] TPM integration

---
 library/oid.c     | 4 ++++
 library/pkwrite.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/library/oid.c b/library/oid.c
index 0a1658f82..0c59c7df6 100644
--- a/library/oid.c
+++ b/library/oid.c
@@ -456,6 +456,10 @@ static const oid_pk_alg_t oid_pk_alg[] =
         { ADD_LEN( MBEDTLS_OID_PKCS1_RSA ),      "rsaEncryption",   "RSA" },
         MBEDTLS_PK_RSA,
     },
+    {
+        { ADD_LEN( MBEDTLS_OID_PKCS1_RSA ),      "rsaEncryption",   "RSA" },
+        MBEDTLS_PK_RSA_ALT,
+    },
     {
         { ADD_LEN( MBEDTLS_OID_EC_ALG_UNRESTRICTED ),  "id-ecPublicKey",   "Generic EC key" },
         MBEDTLS_PK_ECKEY,
diff --git a/library/pkwrite.c b/library/pkwrite.c
index 76159e5a8..6561799aa 100644
--- a/library/pkwrite.c
+++ b/library/pkwrite.c
@@ -63,6 +63,7 @@
 
 #if defined(MBEDTLS_RSA_C)
 #include "mbedtls/rsa.h"
+#include "mbedtls/pk_internal.h"
 #endif
 #if defined(MBEDTLS_ECP_C)
 #include "mbedtls/bignum.h"
@@ -216,6 +217,9 @@ int mbedtls_pk_write_pubkey( unsigned char **p, unsigned char *start,
     if( mbedtls_pk_get_type( key ) == MBEDTLS_PK_RSA )
         MBEDTLS_ASN1_CHK_ADD( len, pk_write_rsa_pubkey( p, start, mbedtls_pk_rsa( *key ) ) );
     else
+    if( mbedtls_pk_get_type( key ) == MBEDTLS_PK_RSA_ALT )
+        MBEDTLS_ASN1_CHK_ADD( len, pk_write_rsa_pubkey( p, start, (mbedtls_rsa_context *)(((mbedtls_rsa_alt_context *)(key->pk_ctx))->key)  ) );
+    else
 #endif
 #if defined(MBEDTLS_ECP_C)
     if( mbedtls_pk_get_type( key ) == MBEDTLS_PK_ECKEY )
-- 
2.17.1

